src/GraphQL/Validation/Complexity/FieldImpactContext.cs

```diff
@@ -0,0 +1,80 @@
+using GraphQL.Execution;
+using GraphQL.Types;
+using GraphQLParser.AST;
+
+namespace GraphQL.Validation.Complexity;
+
+/// <summary>
+/// Provides contextual information about the field being analyzed.
+/// </summary>
+public struct FieldImpactContext
+{
+    /// <summary>
+    /// The <see cref="FieldType"/> definition defined within the graph type specified by <see cref="ParentType"/>, or
+    /// one of the predefined meta fields (__typename, __schema, or __type), which are defined by
+    /// <see cref="ISchema.TypeNameMetaFieldType"/>, <see cref="ISchema.SchemaMetaFieldType"/>, and <see cref="ISchema.TypeMetaFieldType"/>.
+    /// </summary>
+    public FieldType FieldDefinition { get; set; }
+
+    /// <inheritdoc cref="IResolveFieldContext.FieldAst"/>
+    public GraphQLField FieldAst { get; set; }
+
+    /// <summary>
+    /// The field's parent graph type. For meta fields, this defines the parent graph type where the meta field was requested.
+    /// For __typename, this type may be a <see cref="UnionGraphType"/>; otherwise it is an object or interface graph type and
+    /// may be coerced to <see cref="IComplexGraphType"/>.
+    /// </summary>
+    public IGraphType ParentType { get; set; }
+
+    internal ComplexityVisitorContext VisitorContext;
+    private int _parentDepth;
+
+    private IDictionary<string, ArgumentValue>? _arguments;
+    /// <inheritdoc cref="IResolveFieldContext.Arguments"/>
+    public IDictionary<string, ArgumentValue>? Arguments => _arguments ??= (VisitorContext.Arguments?.TryGetValue(FieldAst, out var args) == true ? args : null);
+
+    /// <inheritdoc cref="ResolveFieldContextExtensions.GetArgument{TType}(IResolveFieldContext, string, TType)"/>
+    /// <remarks>Does not coerce the argument name via <see cref="ISchema.NameConverter"/>.</remarks>
+    public TType GetArgument<TType>(string name, TType defaultValue = default!) => Arguments?.TryGetValue(name, out var value) == true ? (TType)value.Value! : defaultValue;
```

Comment:
Alternate logic:
```suggestion
    /// <inheritdoc cref="ResolveFieldContextExtensions.GetArgument{TType}(IResolveFieldContext, string, TType)"/>
    public TType GetArgument<TType>(string name, TType defaultValue = default!)
    {
        // see logic in ResolveFieldContextExtensions.GetArgument
        var arguments = Arguments;
        if (arguments == null)
            return defaultValue;
        var isIntrospectionField = FieldDefinition.Name.StartsWith("__", StringComparison.Ordinal) || ParentType.Name.StartsWith("__", StringComparison.Ordinal);
        var argumentName = isIntrospectionField ? name : (ValidationContext.Schema.NameConverter.NameForArgument(name, (IComplexGraphType)ParentType, FieldDefinition) ?? name);
        if (!arguments.TryGetValue(argumentName, out var value) || value.Value == null)
            return defaultValue;
        if (value is TType typedValue)
            return typedValue;
        var resolvedType = FieldDefinition.Arguments?.Find(argumentName)?.ResolvedType
            ?? throw new InvalidOperationException($"Could not obtain graph type instance for argument '{argumentName}'");
#pragma warning disable IL2087 // Target parameter argument does not satisfy 'DynamicallyAccessedMembersAttribute' in call to target method. The generic parameter of the source method or type does not have matching annotations.
        if (value.Value is IDictionary<string, object?> inputObject)
        {
            return (TType)inputObject.ToObject(typeof(TType), resolvedType);
        }
        return (TType)value.Value.GetPropertyValue(typeof(TType), resolvedType)!;
#pragma warning restore IL2087 // Target parameter argument does not satisfy 'DynamicallyAccessedMembersAttribute' in call to target method. The generic parameter of the source method or type does not have matching annotations.
    }
```

Reply:
This logic mirrors `IResolveFieldContext.GetArgument` exactly, first coercing the argument name via the name converter, then converting to the requested type via `GetPropertyValue`